tensorflow/lite/micro/python/interpreter/src/interpreter_wrapper.cc

```diff
@@ -147,63 +265,56 @@ void InterpreterWrapper::SetInputTensor(PyObject* data, size_t index) {
       /*requirements=*/NPY_ARRAY_CARRAY,
       /*context=*/nullptr));
   if (!array_safe) {
-    PyErr_SetString(PyExc_ValueError, "TFLM cannot convert input to PyArray");
-    return;
+    ThrowValueError("TFLM cannot convert input to PyArray");
   }
 
   PyArrayObject* array = reinterpret_cast<PyArrayObject*>(array_safe.get());
 
   TfLiteTensor* tensor = interpreter_->input(index);
-  if (tensor == nullptr) {
-    PyErr_SetString(PyExc_IndexError,
-                    "Tensor is out of bound, please check tensor index.");
-    return;
-  }
-
-  if (tensor->type == kTfLiteString) {
-    PyErr_SetString(PyExc_ValueError, "TFLM does not support string input.");
-    return;
+  if (!CheckTensor(tensor)) {
+    throw pybind11::error_already_set();
```

Comment:
In the other error checks in this function, we call `Throw*Error()` directly, would it make more sense to just call those in `CheckTensor()` instead of just setting the string in `CheckTensor()` and then `throw pybind11::error_already_set();` here?

Reply:
Good point. Indeed, that was my initial design. However, `CheckTensor` is used in other parts of the code too (like `GetTensorDetails`). To follow TFL, we should return `nullptr` to indicate the error and let the `tflite::PyoOrThrow` handles the throw, unless it is a void function. In that case we call `Throw*Error()' directly. This line is the only exception.

Reply:
Is there any downside to throwing anyway in `GetTensorDetails()` instead of returning a `nullptr`? (and not follow TFL) IMO it's a bit better to keep pybind11 stuff encapsulated in `pybind11_lib.h` now that we have a dedicated file for it, and then we can remove the dependency on `<pybind11.h>` from this file. But it's not too big of a deal now so I'll approve.